subsys/bluetooth/controller/hci/hci.c

```diff
@@ -3624,33 +3630,83 @@ int hci_acl_handle(struct net_buf *buf, struct net_buf **evt)
 #endif /* CONFIG_BT_CONN */
 
 #if CONFIG_BT_CTLR_DUP_FILTER_LEN > 0
-static inline bool dup_found(struct pdu_adv *adv)
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+static void store_adi(int i, struct pdu_adv_adi *adi)
+{
+	if (adi) {
+		memcpy(&dup_filter[i].adi, adi, sizeof(*adi));
+	} else {
+		memset(&dup_filter[i].adi, 0, sizeof(*adi));
+	}
+}
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+
+static bool dup_found(uint8_t adv_type, uint8_t addr_type, uint8_t *addr,
```

Comment:
Would it make sense to use `bt_addr_t` instead of a `uint8_t *` for the address? I guess the comparison further below could be done using the BT addr compare function then.

Reply:
To use `bt_addr_t`, the parameter will need the 6-byte address to be passed by value (copied into address field of bt_addr_t), let alone the added call stack usage for auto-variable in the caller function, this is redundant overhead when the 6-byte value is only used for comparison in the most likely execution path in `dup_found`.

Reply:
> To use bt_addr_t, the parameter will need the 6-byte address to be passed by value

Wouldn't it be possible to copy the data into a struct from the caller, and then use a pointer? That way we actually reduce the number of arguments by one (removing `uint8_t addr_type`). 

It's not a big deal anyhow to keep it as is :)

Reply:
> copy the data into a struct

isnt that passing the parameter by value? arguments are passed using registers which probably/already have the value. Having less code in the caller is always optimal, and compilers do a better job at deciding on parameter passing.

Reply:
Good point :)